### DISCUSSION TITLE
v8.1.0 release - YOLOv8 Oriented Bounding Boxes (OBB)

### DISCUSSION THREAD

**Discussion by glenn-jocher** [ultralytics/ultralytics#7472] [Announcements]
<br>
<img src="https://raw.githubusercontent.com/ultralytics/assets/main/logo/Ultralytics_Logotype_Original.svg" width="320">

# Ultralytics v8.1.0 Release Notes

## Introduction
[Ultralytics](https://ultralytics.com/) proudly announces the v8.1.0 release of [YOLOv8](https://github.com/ultralytics/ultralytics), celebrating a year of remarkable achievements and advancements. This version continues our commitment to making AI technology accessible and powerful, reflected in our latest breakthroughs and improvements.

### 2023 in Review
- **Record-Breaking Engagement:** Over 20 million downloads of the Ultralytics package, with 4 million in December alone! 📈
- **Massive Model Training:** An incredible 19 million YOLOv8 models were trained in 2023, showing the widespread adoption and versatility of our platform. 🌐
- **Diverse Model Usage:** 64% of these models were for object detection, 20% for instance segmentation, 15% for pose estimation, and 1% for image classification. 📊
- **Expanding Global Reach:** YOLOv8 reached 5 million users in 2023 and was run in 15 billion inference jobs across various industries, showcasing its real-world impact. 🌍
- **Documentation in Multiple Languages:** Our docs are now available in 11 languages, catering to our diverse global community. 📚

## Ultralytics v8.1.0 Key Highlights
- **[YOLOv8 OBB Models](https://docs.ultralytics.com/tasks/obb/):** The introduction of Oriented Bounding Box models in YOLOv8 marks a significant step in object detection, especially for angled or rotated objects, enhancing accuracy and reducing background noise in various applications such as aerial imagery and text detection.
- **[Segmentation Support & Enhancements](https://docs.ultralytics.com/tasks/segment/):** Enhanced segmentation capabilities offer more precise image analysis, with improved classification augmentations integrated into Ultralytics training pipelines.
- **Performance Optimizations:** Since our initial release last year we've focused on optimizing every aspect of the YOLOv8 framework, including training, validation, inference, and export, ensuring speed and efficiency without compromising performance.
- **Enhanced Model Architecture & Training Features:** Incremental updates in model architecture, training features, and dataset support, including integration with Open Images V7 dataset and improved image classification models.
- **API and CLI Improvements:** Enhanced user experience with refined API and CLI, including the [Ultralytics Explorer](https://docs.ultralytics.com/datasets/explorer/dashboard/) tool for advanced dataset exploration and interaction.
- **[PaddlePaddle, NCNN, PNNX, TensorRT & Other Integrations](https://docs.ultralytics.com/modes/export/):** Strengthened integration with multiple other platforms, offering users more deployment flexibility and compatibility for YOLOv8 users.
- **Diverse Contributions & Ultralytics HUB Evolution:** The integration of over **1000 pull requests by 230 contributors** and the growth of [Ultralytics HUB](https://hub.ultralytics.com), with it's own series of version updates, highlights the community's vital role in the development of YOLOv8.

## Community Engagement and Support
- **[Expanding Documentation](https://docs.ultralytics.com):** Our documentation now spans 11 languages, with over 200 pages, providing comprehensive guides for various real-world applications.
- **[Custom-Trained YOLOv8 Models](https://docs.ultralytics.com/modes/train/):** With the ability to train models on custom data, 19 million YOLOv8 models were trained in 2023 alone, catering to diverse needs across object detection, segmentation, pose estimation, and image classification.
- **[User Contributions](https://github.com/ultralytics/ultralytics/graphs/contributors):** We encourage and appreciate user-contributed examples and stories, showcasing the versatility and real-world impact of YOLOv8.

## Summary
Ultralytics v8.1.0 is a testament to a year of innovation, with the integration of Oriented Object Detection, enhanced classification models, and a strong focus on user experience and community engagement. We thank our users and contributors for their invaluable support and look forward to another year of groundbreaking advancements in the field of AI and computer vision in 2024! 🌟🚀🎉

**Comment by pahaht** [comment]
Thank you

**Reply by pderrenger** [reply]
@pahaht you're welcome! If you have any questions or need further assistance, feel free to reach out. Happy coding with YOLOv8! 😊🚀

**Comment by Wang-taoshuo** [comment]
Q1: I directly input !pip install ultralytics==8.0.28 to !pip install ultralytics==8.1.0 in an ipynb file. After updating, the versions of Torch and CUDA, etc., should not be changed, right?

Q2: After the update, have you resolved the issue of seg_loss: nan?
![螢幕擷取畫面 2024-03-06 191832](https://github.com/ultralytics/ultralytics/assets/119144468/50ed1c86-f287-406e-9409-7beff97b574f)

**Reply by glenn-jocher** [reply]
Hello! 😊

Q1: Correct, updating the `ultralytics` package from 8.0.28 to 8.1.0 shouldn't change your Torch or CUDA versions. These dependencies are managed separately, so you're all set there!

Q2: Yes, we've addressed the `seg_loss: nan` issue in the 8.1.0 release. If you're still encountering this problem after updating, please ensure your dataset annotations are correct and consider adjusting the learning rate or segmentation loss weights. If the issue persists, feel free to share more details, and we'll be happy to help further!

Happy coding! 🚀

**Reply by Wang-taoshuo** [reply]
Thank you~

**Reply by glenn-jocher** [reply]
@Wang-taoshuo of course! Please share the specific GitHub issue comment you'd like me to respond to, and I'll craft a concise and friendly reply for you. 😊

**Comment by fluffytid** [comment]
Hi, I've been working for 11 months on a project with playing cards, it would make a lot of sense to have an OBB model instead, but all my training images are labeled as squares with labelimg. Is there a new tool for training with oriented boxes now?. I think my model is far too deep into squared recognition to turn into OBB, and it scares me a lot to move now after all this work invested.

**Reply by glenn-jocher** [reply]
@fluffytid hi there!

Great to hear about your dedication to your project! Yes, transitioning to an Oriented Bounding Box (OBB) model could indeed provide better results for objects like playing cards that have a specific orientation. For labeling images with oriented boxes, many users find success with CVAT, which supports OBB annotations.

Transitioning from square to OBB annotations is a significant step but it might not be as daunting as it seems. Your already labeled dataset still holds value, especially when transitioning. While direct conversion tools are scarce, you could potentially automate the conversion of square labels to a rough OBB format and refine as needed.

Here's a pseudo-code to get you started on converting square bounding boxes to OBB:

```python
def square_to_obb(x_center, y_center, width, height, angle=0):
    # Your method to convert square bbox to obb might vary
    return (x_center, y_center, width, height, angle)
```

Remember, transitioning to a new model type is a step-by-step journey. Testing with a small subset of your data can help you slowly move towards full OBB implementation without risking the solid foundation you've built over the months. 🚀

Don't hesitate to share your progress or ask more questions. We're here to support you at every step of this exciting journey!

**Comment by fluffytid** [comment]
Thank you, I already did some data augmentation algorithm based on
contours, so I guess I could, still pretty scary to fix something that is
working.

My results so far are good enough, no need to get better precision, my main
problem at the moment is performance, are there any tests regarding if OBB
is faster or slower than 8.0?

**Reply by glenn-jocher** [reply]
Hey there!

Happy to help with your performance concerns! 🚀 When it comes to OBB (Oriented Bounding Box) performance compared to YOLOv8.0, there's no one-size-fits-all answer. It mainly depends on your specific use case and data. Generally, OBB might introduce a slight overhead due to the additional parameters (orientation angle) but can be more efficient for detection if your objects significantly benefit from orientation information.

For your current performance concerns, I'd recommend exploring model optimization techniques if you haven't already. This can involve:
- Switching to a lighter model architecture if you're using a heavy one.
- Experimenting with quantization or model pruning if precision trade-offs are acceptable.
- If you're running inference on a GPU, leveraging TensorRT for model conversion can significantly boost inference speed.

For a quick example on switching to a lighter model:
```bash
yolo export model=yolov8n.pt format=onnx  # Standard export
yolo export model=yolov8s.pt format=onnx  # Exporting a smaller model variant
```
And remember, optimizing model performance often involves balancing speed, accuracy, and computational resources, so it’s all about finding the right mix for your needs.

If there are specific areas you're targeting for performance improvement (e.g., inference time, model size), feel free to share more details. We're here to help guide you through! 😊

**Comment by phoenix-JM** [comment]
is yolov8-obb now yolov11-obb after yolo v11 release??

**Reply by glenn-jocher** [reply]
@phoenix-JM yOLOv8-OBB remains as YOLOv8-OBB; it hasn't changed with the YOLO11 release.

**Reply by phoenix-JM** [reply]
Every link that previously leads to yolov8-obb now opens yolov11-obb weights and models.

**Reply by glenn-jocher** [reply]
YOLOv8-OBB and YOLO11-OBB are separate models; please ensure you're accessing the correct links for each version.

**Reply by phoenix-JM** [reply]
Thank you.

**Reply by glenn-jocher** [reply]
You're welcome! If you have any more questions, feel free to ask.

**Comment by Petros626** [comment]
@glenn-jocher https://github.com/ultralytics/ultralytics/issues/189#issuecomment-2646438342 could you imagine to update the OBB head to this improvement?

**Reply by glenn-jocher** [reply]
Thanks for the suggestion—yes, we’re open to OBB head improvements; please submit a PR against main with the change plus DOTAv1 mAP/speed benchmarks and a brief ablation following the Ultralytics Contributing Guide, and we’ll review promptly.

**Reply by Petros626** [reply]
It was actually intended as a request to you, as I am not in a position to implement something like this. I think it would definitely be worth a try, considering the results in the paper (https://www.mdpi.com/1424-8220/24/17/5850).

**Reply by glenn-jocher** [reply]
Thanks for flagging this—understood; we can’t commit to implementing it right now, but if you open a Feature Request summarizing the method and expected gains, plus any reproducible DOTAv1 accuracy/latency evidence, we’ll triage based on impact and community interest, with any OBB head changes targeting YOLO11-OBB first.
